src/postamble.js

```diff
@@ -452,9 +449,6 @@ function exit(status, implicit) {
 function procExit(code) {
   EXITSTATUS = code;
   if (!keepRuntimeAlive()) {
-#if USE_PTHREADS
-    PThread.terminateAllThreads();
-#endif
```

Comment:
I think this `terminateAllThreads` is actually still useful for the case when `_exit` or `_Exit` is called with multiple threads running.  

On node is probably doesn't matter since `quit_t` will bring down the whole process, but on the web it will mean we don't leak workers in this case.

This will always happen after any atexit processing has occurred.

Reply:
I thought it was redundant to call `terminateAllThreads` here, since the only callers of `procExit` are:
https://github.com/emscripten-core/emscripten/blob/565fb3651ed185078df1a13b8edbcb6b2192f29e/src/postamble.js#L449
(which will be called just after `exitRuntime`, in case we don't need to keep the runtime alive)

And:
https://github.com/emscripten-core/emscripten/blob/cecc2feb007726dfacc0dc86e8493364234be1de/src/library_wasi.js#L14
(which will be only called by `MINIMAL_RUNTIME` and `STANDALONE_WASM`, AFAIK)

I added the `terminateAllThreads` in `postamble_minimal.js` to make sure that it still terminates all threads during `exit` when linking with `-sMINIMAL_RUNTIME`. But for `STANDALONE_WASM` pthreads are not supported at all, so this seemed redundant.
https://github.com/emscripten-core/emscripten/blob/565fb3651ed185078df1a13b8edbcb6b2192f29e/emcc.py#L2060-L2061

Reply:
> I thought it was redundant to call `terminateAllThreads` here, since the only callers of `procExit` are:
> https://github.com/emscripten-core/emscripten/blob/565fb3651ed185078df1a13b8edbcb6b2192f29e/src/postamble.js#L449
> 
> 
> (which will be called just after `exitRuntime`, in case we don't need to keep the runtime alive)
> And:
> https://github.com/emscripten-core/emscripten/blob/cecc2feb007726dfacc0dc86e8493364234be1de/src/library_wasi.js#L14
> 
> 
> (which will be only called by `MINIMAL_RUNTIME` and `STANDALONE_WASM`, AFAIK)

This callsite is used if the user calls `_exit` or `_Exit`.   Its basically the exit syscall.  Its not just used in STANDALONE_WASM IIUC.

Reply:
Ah, yes. I missed this one:
https://github.com/emscripten-core/emscripten/blob/b1d2d393009ff403c26b080a686895e0e60a68cd/system/lib/libc/musl/src/exit/_Exit.c#L7
(where `__wasi_proc_exit` is just an alias for `proc_exit`)

Commit ea90c63dea2d1fbec29c4b727c09b6ed653b9306 fixes this.

Reply:
Ah, this also seems to fix those `asan.*thread*` failures without having to change `_free` to `_emscripten_builtin_free` in `PThread.freeThreadData`. So, I've reverted that with commit https://github.com/emscripten-core/emscripten/commit/0f6f592f4ad9b6ccf28336dad8e618935e77df1b.